packages/jira-adapter/src/filters/sla_addition_deployment.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ *                      Copyright 2024 Salto Labs Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  AdditionChange,
+  Change,
+  InstanceElement,
+  getChangeData,
+  isAdditionChange,
+  isInstanceChange,
+  toChange,
+} from '@salto-io/adapter-api'
+import { createSchemeGuard, getParent } from '@salto-io/adapter-utils'
+import _ from 'lodash'
+import { logger } from '@salto-io/logging'
+import { elements as elementUtils, config as configDeprecated } from '@salto-io/adapter-components'
+import Joi from 'joi'
+import { defaultDeployChange, deployChanges } from '../deployment/standard_deployment'
+import { FilterCreator } from '../filter'
+import { SLA_TYPE_NAME } from '../constants'
+import JiraClient from '../client/client'
+
+const log = logger(module)
+const { replaceInstanceTypeForDeploy } = elementUtils.ducktype
+
+type SlaParams = {
+  id: number
+  name: string
+}
+
+type SlaGetResponse = {
+  timeMetrics: SlaParams[]
+}
+const SLA_RESPONSE_SCHEME = Joi.object({
+  timeMetrics: Joi.array()
+    .items(
+      Joi.object({
+        id: Joi.number().required(),
+        name: Joi.string().required(),
+      }).unknown(true),
+    )
+    .required(),
+})
+  .unknown(true)
+  .required()
+
+const isSlaResponse = createSchemeGuard<SlaGetResponse>(SLA_RESPONSE_SCHEME)
+
+const getExistingSlaNamesAndIds = async (
+  parent: InstanceElement,
+  client: JiraClient,
+): Promise<(string | number)[][]> => {
+  try {
+    const response = await client.get({
+      url: `/rest/servicedesk/1/servicedesk/agent/${parent.value.key}/sla/metrics`,
+    })
+    if (!isSlaResponse(response.data)) {
+      return []
+    }
+    const existingSlas = response.data.timeMetrics.map(sla => [sla.name, sla.id])
```

Comment:
I think it's more safe to work with types objects
```suggestion
    const existingSlas = response.data.timeMetrics.map(({ name, id }) => ({ name, id }))
```